### PR TITLE
feat: new clients, convert to TypeScript

### PR DESCRIPTION
**Pull request by alexander-fenster**
So... we have a bug in gapic-generator that breaks the proto updates for this repo (https://github.com/googleapis/nodejs-datacatalog/pull/60). Instead of fixing that bug there, I suggest that we just move this library to TypeScript :)  (since the TypeScript generator does not have this problem)

This PR brings in new clients (`PolicyTagManagerClient` and `PolicyTagManagerSerializationClient` in addition to the existing `DataCatalogClient`) and converts the library to TypeScript.  No incompatible changes are expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,28 +28,39 @@
...
+    "gts": "^1.1.0",
     "jsdoc": "^3.6.2",
     "jsdoc-fresh": "^1.0.1",
     "jsdoc-region-tag": "^1.0.2",
     "linkinator": "^1.4.0",
     "mocha": "^6.1.4",
-    "prettier": "^1.17.0"
+    "pack-n-play": "^1.0.0-2",
+    "prettier": "^1.17.0",
+    "typescript": "~3.5.0"
```
Comment (JustinBeckwith) [inline package.json]: This should go straight to `3.7.0`.  Do we have renovate running on `gapic-generator-typescript`?

File: package.json [hunk 2]
```diff
@@ -28,28 +28,39 @@
...
-    "test": "c8 mocha test/*.js",
+    "fix": "gts fix && eslint . --fix",
+    "lint": "gts check && eslint .",
+    "test": "c8 mocha build/test/*.js",
     "predocs-test": "npm run docs",
     "docs-test": "linkinator docs",
     "samples-test": "c8 mocha samples/test/*.js",
-    "system-test": "c8 mocha --timeout=5000 system-test/*.js"
+    "system-test": "c8 mocha --timeout=5000 build/system-test/*.js",
+    "check": "gts check && eslint .",
```
Comment (JustinBeckwith) [inline package.json]: Just drop `check` and `clean` YAGNI

File: synth.py [hunk 3]
```diff
@@ -22,51 +22,34 @@
...
- * @namespace google.type
- */
-""")
-
-s.replace("src/v1beta1/doc/google/cloud/datacatalog/v1beta1/doc_gcs_fileset_spec.js",
-r"""\* @see \[google.cloud.datacatalog.v1beta1.GcsFileSpec definition in proto format\]{@link https://github.com/googleapis/googleapis/blob/master/google/cloud/datacatalog/v1beta1/gcs_fileset_spec.proto}
-""",
-r"""
-""")
+# Fix broken links to cloud.google.com documentation
```
Comment (JustinBeckwith) [inline synth.py]: Please file a docs bug on the repository to come back around and clean this up

File: synth.py [hunk 4]
```diff
@@ -22,51 +22,34 @@
...
+s.replace('src/v1beta1/*.ts', '/data-catalog/docs/', 'https://cloud.google.com/data-catalog/docs/')
 
 # Node.js specific cleanup
 subprocess.run(['npm', 'install'])
 subprocess.run(['npm', 'run', 'fix'])
+subprocess.run(['npx', 'compileProtos', 'src'])
+
+# Fix broken links to cloud.google.com documentation
+s.replace('protos/protos.js', '/data-catalog/docs/', 'https://cloud.google.com/data-catalog/docs/')
```
Comment (JustinBeckwith) [inline synth.py]: Same as above

File: system-test/.eslintrc.yml [hunk 5]
```diff
@@ -2,4 +2,4 @@
 env:
   mocha: true
 rules:
-  no-console: off
\ No newline at end of file
+  no-console: off
```
Comment (JustinBeckwith) [inline system-test/.eslintrc.yml]: Could we drop these two lines, and just delete the console logs from the system test?

File: package.json (hunk 1 above)
Comment (alexander-fenster) [reply package.json]: Huh. This is an existing repo so `package.json` is not coming from the generator. I just did `npx gts init` here and turns out it brings some outdated stuff. Fixing now

File: system-test/.eslintrc.yml (hunk 5 above)
Comment (alexander-fenster) [reply system-test/.eslintrc.yml]: Yes! https://github.com/googleapis/gapic-generator-typescript/pull/153

File: synth.py (hunk 4 above)
Comment (alexander-fenster) [reply synth.py]: https://github.com/googleapis/nodejs-datacatalog/issues/62

File: synth.py (hunk 3 above)
Comment (alexander-fenster) [reply synth.py]: https://github.com/googleapis/nodejs-datacatalog/issues/62

File: package.json (hunk 2 above)
Comment (alexander-fenster) [reply package.json]: Done!

**Pull request merged**